npm/pkg/controlplane/controllers/v1/podController.go

```diff
@@ -82,6 +82,16 @@ func (nPod *NpmPod) updateNpmPodAttributes(podObj *corev1.Pod) {
 	}
 }
 
+// noUpdate evaluates whether NpmPod is required to be update given podObj.
+func (nPod *NpmPod) noUpdate(podObj *corev1.Pod) bool {
+	return nPod.Namespace == podObj.ObjectMeta.Namespace &&
+		nPod.Name == podObj.ObjectMeta.Name &&
+		nPod.Phase == podObj.Status.Phase &&
+		nPod.PodIP == podObj.Status.PodIP &&
+		reflect.DeepEqual(nPod.Labels, podObj.ObjectMeta.Labels) &&
```

Comment:
Please don't import `reflect` outside of tests unless you're doing meta-programming. It's expensive to use, and the result of `DeepEqual` specifically is sometimes (at the worst times) not what you would expect.
NPM can't spare this performance 🥲

Reply:
@rbtr Good catch. We also recently found performance issues using `reflect`. I think this is good time to revise it in this PR.